src/python/zquantum/core/circuit/_gate.py

```diff
@@ -888,3 +915,116 @@ def _unitary_(self):
 
     def __str__(self):
         return f"Db({self.p})"
+
+
+class MCTGate(object):
+    """
+    An Orquestra gate class that deals with synthesis of a Multi Toffoli Gate
+    Although this inherits the Gate class it has extra attributes fundamentally related to
+    the fact that the Multi Toffoli Gate is a circuit rather a gate applied natively on a device
+
+    Attributes:
+    circuit (Orquestra Circuit): Needed to apply the mct gate from qiskit
+
+    control_qubits (list): This list contains integer labels for the qubits thats will act as the
+    control qubits. e.g [1, 2, 3, 4]
+    target_qubits (list): This is a list that contains integer labels for the qubits that are target qubits
+
+    all_qubits (list): A list of zmachine.core.qubit.Qubit This is necessary to determine what qubits
+    are the ancilla
+
+    ancilla_qubits (list): A list that contians integer labels for all the qubits labels that are the ancilla. These are needed since the MCT synthesis
+    requires in general addition of ancilla qubits
+
+    ccx_list (list): A list that contains CCX orquestra gates that represent the decompostion of the MCT gate
+
+    ccx_decomposition (qiskit circuit): Contains the Toffoli Decomposition which is the output from the mct gate
+
+    qiskit_ctrl_q (list): List of qiskit qubits that are control qubits for the MCT gate
+    qiskit_targ_q (list): List of qiskit qubits that are targ qubits for the MCT gate
+
+    qiskit_ancilla_q (list): List of qiskit qubits that are ancilla qubits for the mct gate synthesis
+
+
+    """
+
+    def __init__(self, circuit, control_qubits=[], target_qubits=[]):
```

Comment:
You should not use empty lists as default parameters – it might lead to weird behaviour and very hard to track bugs. It is much better to do:
```
def __init__(self, control_qubits=None):
    if control_qubits is None:
        control_qubits = []
```